packages/twenty-server/src/core/auth/controllers/google-gmail-auth.controller.ts

```
@@ -37,14 +37,21 @@ export class GoogleGmailAuthController {
     const { workspaceMemberId, workspaceId } =
       await this.tokenService.verifyTransientToken(transientToken);
 
-    await this.googleGmailService.saveConnectedAccount({
-      handle: email,
-      workspaceMemberId: workspaceMemberId,
-      workspaceId: workspaceId,
-      provider: 'gmail',
-      accessToken,
-      refreshToken,
-    });
+    const demoWorkspaceIds = this.environmentService.getDemoWorkspaceIds();
+
+    if (demoWorkspaceIds.includes(workspaceId)) {
+      throw new Error('Cannot connect Gmail account to demo workspace');
+    }
+
+    if (workspaceId)
```

should we throw if workspaceId is not defined or null?